src/main/java/uk/gov/hmcts/reform/authorisation/ServiceAuthorisationApi.java

```diff
@@ -0,0 +1,15 @@
+package uk.gov.hmcts.reform.authorisation;
+
+import org.springframework.cloud.netflix.feign.EnableFeignClients;
+import org.springframework.cloud.netflix.feign.FeignClient;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@EnableFeignClients
+@FeignClient(name = "idam-s2s-auth", url = "${idam.s2s-auth.url}")
```

Comment:
Allow users of this lib to instantiate the client with passed url. If I understand correctly, this is taked from env vars?

Also, add sample usage to README.

Reply:
how?
this is where the feign magic comes in, set a url in your properties / env vars and it will be resolved.

Reply:
something like
```
ServiceAuthorisationApi api = 
  Feign
    .builder()                 
    .target(ServiceAuthorisationApi.class, "https://whatever.com");

```

Reply:
that should probably work already?
but when we have someone with that use case it can be looked at.
Currently all apps I am aware of use spring boot, so not sure why we would try and do that.

Reply:
Maybe, but all the Feign stuff should be internal to this library imho.

I think it would be much clearer if spring boot apps using this library would define a bean which in their configuration where this class is created with explicitly passed url (albeit retrieved from env vars :))

Reply:
not too worried either way here.